cdci_data_analysis/plugins/importer.py

```diff
@@ -55,16 +55,20 @@
 }
 
 instrument_factory_list = []
+instrument_name_dict = {}
 # pre-load the empty instrument factory
 
 instrument_factory_list.append(empty_instrument.my_instr_factory)
 instrument_factory_list.append(empty_async_instrument.my_instr_factory)
+instrument_name_dict['empty'] = empty_instrument.my_instr_factory
+instrument_name_dict['empty-async'] = empty_async_instrument.my_instr_factory
 
 for plugin_name in cdci_plugins_dict:
     logger.info("found plugin: %s", plugin_name)
 
     try:
         e = importlib.import_module(plugin_name+'.exposer')
+        instrument_name_dict.update(e.instr_name_dict)
```

Comment:
this changes dispatcher-plugin interface, breaking it for existing plugins. There should be a reasonable default, compatible with what was before, for plugins which do not implement this.

Reply:
indeed, I need also a small change in the various plugins

Reply:
if we don't want to change the plugins interface, then this is not a solution